route/route.go

```diff
@@ -330,6 +375,117 @@ func (r *Router) batch(w http.ResponseWriter, req *http.Request) {
 	w.Write(response)
 }
 
+func (r *Router) Export(ctx context.Context, req *collectortrace.ExportTraceServiceRequest) (*collectortrace.ExportTraceServiceResponse, error) {
+
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		r.Logger.Error().Logf("Unable to retreive metadata from OTLP request.")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	var requestID types.RequestIDContextKey
+
+	apiKey, dataset := getAPIKeyAndDatasetFromMetadata(md)
+	if apiKey == "" {
+		r.Logger.Error().Logf("Received OTLP request without Honeycomb APIKey header")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+	if dataset == "" {
+		r.Logger.Error().Logf("Received OTLP request without Honeycomb dataset header")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	apiHost, err := r.Config.GetHoneycombAPI()
+	if err != nil {
+		r.Logger.Error().Logf("Unable to retrieve APIHost from config while processing OTLP batch")
+		return &collectortrace.ExportTraceServiceResponse{}, nil
+	}
+
+	for _, resourceSpan := range req.ResourceSpans {
+		resourceAttrs := make(map[string]interface{})
+
+		if resourceSpan.Resource != nil {
+			addAttributesToMap(resourceAttrs, resourceSpan.Resource.Attributes)
+		}
+
+		for _, librarySpan := range resourceSpan.InstrumentationLibrarySpans {
+			library := librarySpan.InstrumentationLibrary
+			if library != nil {
+				if library.Name != "" {
```

Comment:
nit, but for clarity, we should stick to checking for an empty string or checking that the length is greater than 0, not both. I vote `if len(library.Name) > 0`.